packages/block-editor/src/components/responsive-block-control/index.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import { uniqueId } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+import { Fragment, useState } from '@wordpress/element';
+
+import {
+	ToggleControl,
+} from '@wordpress/components';
+
+export function ResponsiveBlockControlLabel( { property, device } ) {
+	const accessibleLabel = sprintf( __( 'Controls the %s property for %s devices.' ), property, device );
+	const descId = uniqueId();
+	return (
+		<Fragment>
+			<span aria-describedby={ `rbc-desc-${ descId }` }>
+				{ device }
+			</span>
+			<span className="screen-reader-text" id={ `rbc-desc-${ descId }` }>{ accessibleLabel }</span>
+		</Fragment>
+	);
+}
+
+function ResponsiveBlockControl( props ) {
+	const { legend, property, toggleLabel, responsiveControlsActive = false, renderDefaultControl, defaultLabel = __( 'All' ), devices = [ __( 'Desktop' ), __( 'Tablet' ), __( 'Mobile' ) ], renderResponsiveControls } = props;
+
+	const [ isResponsiveMode, setIsResponsiveMode ] = useState( responsiveControlsActive );
+
+	if ( ! legend || ! property || ! renderDefaultControl ) {
+		return null;
+	}
+
+	const toggleControlLabel = toggleLabel || sprintf( __( 'Use the same %s on all screensizes.' ), property );
+
+	const toggleHelpText = __( 'Toggle between using the same value for all screen sizes or using a unique value per screen size.' );
+
+	const defaultControl = renderDefaultControl( defaultLabel );
+
+	const defaultResponsiveControls = () => {
+		return devices.map( ( deviceLabel, index ) => (
+			<Fragment key={ index }>
+				{ renderDefaultControl( deviceLabel ) }
```

Comment:
We only pass the label to the renderDefaultControl, in a real case if we were storing the value somewhere e.g: attribute we would need to know an identifier of the size we are referring to. So, besides passing a label I think we need to pass something like a device id.

Reply:
Makes sense. I now pass the entire `device` object to the callback to ensure it has access.